orchestrator/careplanservice/integration_test.go

```diff
@@ -42,29 +44,58 @@ func Test_Integration_Service_handleCreateTask(t *testing.T) {
 			require.NotNil(t, createdCarePlan.Id)
 			require.Len(t, createdCareTeams, 1, "expected 1 CareTeam")
 			require.NotNil(t, createdCareTeams[0].Id)
-		})
-		t.Run("Create Task", func(t *testing.T) {
-			task := fhir.Task{
-				BasedOn: []fhir.Reference{
-					{
-						Type:      to.Ptr("CarePlan"),
-						Reference: to.Ptr("CarePlan/" + *carePlan.Id),
+
+			t.Run("Create Task", func(t *testing.T) {
+				task = fhir.Task{
+					BasedOn: []fhir.Reference{
+						{
+							Type:      to.Ptr("CarePlan"),
+							Reference: to.Ptr("CarePlan/" + *carePlan.Id),
+						},
 					},
-				},
-			}
-			err := carePlanContributor.Create(task, &task)
-			require.NoError(t, err)
+					Requester: coolfhir.LogicalReference("Organization", coolfhir.URANamingSystem, "1"),
+					Owner:     coolfhir.LogicalReference("Organization", coolfhir.URANamingSystem, "2"),
+				}
 
-			t.Run("Check Task properties", func(t *testing.T) {
-				require.NotNil(t, task.Id)
-				require.Equal(t, "CarePlan/"+*carePlan.Id, *task.BasedOn[0].Reference, "Task.BasedOn should reference CarePlan")
-			})
-			t.Run("Check that CarePlan.activities contains the Task", func(t *testing.T) {
-				err = carePlanContributor.Read("CarePlan/"+*carePlan.Id, &carePlan)
+				err := carePlanContributor.Create(task, &task)
 				require.NoError(t, err)
-				require.Len(t, carePlan.Activity, 1)
-				require.Equal(t, "Task", *carePlan.Activity[0].Reference.Type)
-				require.Equal(t, "Task/"+*task.Id, *carePlan.Activity[0].Reference.Reference)
+
+				t.Run("Check Task properties", func(t *testing.T) {
+					require.NotNil(t, task.Id)
+					require.Equal(t, "CarePlan/"+*carePlan.Id, *task.BasedOn[0].Reference, "Task.BasedOn should reference CarePlan")
+				})
+				t.Run("Check that CarePlan.activities contains the Task", func(t *testing.T) {
+					err = carePlanContributor.Read("CarePlan/"+*carePlan.Id, &carePlan)
+					require.NoError(t, err)
+					require.Len(t, carePlan.Activity, 1)
+					require.Equal(t, "Task", *carePlan.Activity[0].Reference.Type)
+					require.Equal(t, "Task/"+*task.Id, *carePlan.Activity[0].Reference.Reference)
+				})
+
+				t.Run("Accept Task", func(t *testing.T) {
```

Comment:
Should this not be in method `Test_Integration_Service_handleUpdateTask`? 

Otherwise, if it is more convenient to test the update after the create, maybe we can rename the test method `Test_Integration_Service_handleCreateTask` something more generic/applicable like `Test_Integration_Service_handleCRUDTask` or similar?

Reply:
This had become a bit messy, I'd like to have 1 integration test for Task lifecycle (for now), so I'll just skip on nesting tests, and just make it 1 big test.

Reply:
Sounds good